Issue #63: Add /api/users/self endpoint

lmorchard opened the pull request:
- Add /api/users/self API endpoint to expose data about the currently
  logged in user. (i.e. user ID, installed experiments)
- Move sadface JS data into template markup
- Add the user and addon props to overall Me model
- Frontend client tweaks to use the API
- Add .editorconfig to idea_town/frontend/static-src so my editor can do
  different indenting in Python & JS
- Server-side tests

johngruen commented:
@meandavejustice 
![giphy](https://cloud.githubusercontent.com/assets/3323249/9188548/93ac20fa-3fab-11e5-838c-078f7096312c.gif)

lmorchard committed:
Issue #63: Add /api/users/self endpoint

* Add /api/users/self API endpoint to expose data about the currently
  logged in user. (i.e. user ID, installed experiments)

* Move sadface JS data into template markup

* Add the user and addon props to overall Me model

* Frontend client tweaks to use the API

* Add .editorconfig to idea_town/frontend/static-src so my editor can do
  different indenting in Python & JS

* Server-side tests
